pkg/kubefed2/join.go

```diff
@@ -757,18 +867,31 @@ func populateSecretInHostCluster(clusterClientset, hostClientset client.Interfac
 		v1Secret.Name = secretName
 	}
 
-	v1SecretResult, err := hostClientset.CoreV1().Secrets(namespace).Create(&v1Secret)
-	if err != nil {
-		if idempotent && errors.IsAlreadyExists(err) {
-			glog.V(2).Infof("Secret %s already exists in host cluster", secretName)
-		} else {
-			glog.V(2).Infof("Could not create secret in host cluster: %v", err)
+	if secretName != "" {
+		existingSecret, err := hostClientset.CoreV1().Secrets(namespace).Get(secretName, metav1.GetOptions{})
+		switch {
+		case err != nil && !errors.IsNotFound(err):
+			glog.V(2).Infof("Could not get secret %s in host cluster: %v", secretName, err)
 			return nil, err
+		case err == nil && errorOnExisting:
+			return nil, fmt.Errorf("secret %s already exists in host cluster", secretName)
+		case err == nil:
+			existingSecret.Data = v1Secret.Data
```

Comment:
@marun sorry I'm just getting back to this PR today. I still think that overwriting the secret is the right thing to do since the service account on the target cluster could have been (re-)created and you have a different token. To mitigate risk of overwriting stuff we could merge the Data map, but not sure that is a better result.

Reply:
To be clear, I'm worried because join can be configured with an arbitrary secret name, and a secret is a generic type rather than federation-specific.  It would be entirely possible, then, for join to unintentionally override an arbitrary secret.  Maybe an existing secret should result in an error unless the name is the default (i.e. not an arbitrary name provided by the user)?

Reply:
That sounds fine. I'll change it. I should hopefully submit an updated (hopefully cleaner) PR today.